src/MoBi.Presentation/Tasks/Edit/EditTaskFor.cs

```diff
@@ -42,22 +50,59 @@ public abstract class EditTaskFor<T> : IEditTaskFor<T> where T : class, IObjectB
       protected readonly IMoBiApplicationController _applicationController;
       protected readonly IInteractionTask _interactionTask;
       protected readonly IInteractionTaskContext _interactionTaskContext;
+      private readonly IMoBiXmlSerializerRepository _xmlSerializerRepository;
+      private readonly IContainer _container;
+      private readonly IDimensionFactory _dimensionFactory;
+      private readonly IObjectBaseFactory _objectBaseFactory;
+      private readonly ICloneManagerForModel _cloneManagerForModel;
       public string ObjectName { get; private set; }
 
-      protected EditTaskFor(IInteractionTaskContext interactionTaskContext)
+      protected EditTaskFor(
+         IInteractionTaskContext interactionTaskContext,
+         IMoBiXmlSerializerRepository xmlSerializerRepository = null, 
+         IContainer container = null,
+         IDimensionFactory dimensionFactory = null,
+         IObjectBaseFactory objectBaseFactory = null,
+         ICloneManagerForModel cloneManagerForModel = null)
       {
          _interactionTaskContext = interactionTaskContext;
          _applicationController = interactionTaskContext.ApplicationController;
          _interactionTask = interactionTaskContext.InteractionTask;
          _context = interactionTaskContext.Context;
          ObjectName = _interactionTaskContext.GetTypeFor<T>();
+         _xmlSerializerRepository = xmlSerializerRepository;
+         _container = container;
+         _dimensionFactory = dimensionFactory;
+         _objectBaseFactory = objectBaseFactory;
+         _cloneManagerForModel = cloneManagerForModel;
       }
 
       public virtual void Edit(T objectToEdit)
       {
          _context.PublishEvent(new EntitySelectedEvent(objectToEdit, this));
       }
 
+      protected TEntity Clone<TEntity>(TEntity entityToClone) where TEntity : class
```

Comment:
@abdelr This is not correct. A clone method like this one will keep all internal ids resulting in different object with same id. I know this is a draft but you cannot rely on this method

Reply:
I open the PR by mistake. Let me know when you want me to review